Bottom: da512ddcdb47103978a99c0875512aa9d2d06e32
Top:    46f9217888c9acebd399b8bae5da7220e1c019e9
Author: Robert Collins <robertc@robertcollins.net>
Date:   2013-02-17 14:12:22 +1300

Refresh of copy-stream

---

diff --git a/testtools/testresult/real.py b/testtools/testresult/real.py
index 382e872..1d4aff9 100644
--- a/testtools/testresult/real.py
+++ b/testtools/testresult/real.py
@@ -374,6 +374,10 @@ class StreamResult(object):
         """
 
 
+def domap(*args, **kwargs):
+    return list(map(*args, **kwargs))
+
+
 class CopyStreamResult(object):
     """Copies all event it receives to multiple results.
     
@@ -384,19 +388,19 @@ class CopyStreamResult(object):
         self.targets = targets
 
     def startTestRun(self):
-        map(methodcaller('startTestRun'), self.targets)
+        domap(methodcaller('startTestRun'), self.targets)
 
     def stopTestRun(self):
-        map(methodcaller('stopTestRun'), self.targets)
+        domap(methodcaller('stopTestRun'), self.targets)
 
     def estimate(self, *args, **kwargs):
-        map(methodcaller('estimate', *args, **kwargs), self.targets)
+        domap(methodcaller('estimate', *args, **kwargs), self.targets)
 
     def file(self, *args, **kwargs):
-        map(methodcaller('file', *args, **kwargs), self.targets)
+        domap(methodcaller('file', *args, **kwargs), self.targets)
 
     def status(self, *args, **kwargs):
-        map(methodcaller('status', *args, **kwargs), self.targets)
+        domap(methodcaller('status', *args, **kwargs), self.targets)
 
 
 class MultiTestResult(TestResult):
